cores/arduino/HardwareTimer.cpp

```diff
@@ -908,9 +908,9 @@ void HardwareTimer::setInterruptPriority(uint32_t preemptPriority, uint32_t subP
   * @param  callback: interrupt callback
   * @retval None
   */
-void HardwareTimer::attachInterrupt(void (*callback)(HardwareTimer *))
+void HardwareTimer::attachInterrupt(callback_function_t callback)
 {
-  if (callbacks[0] != NULL) {
+  if (callbacks[0] != nullptr) {
```

Comment:
I would remove the `!- nullptr` here, just let the callback be implicitly converted to `bool` by the if. This does exactly the same, but is a bit more future-proof (support for `!= null` [will be removed in C++20](https://en.cppreference.com/w/cpp/utility/functional/function/operator_cmp) for some reason) and does not suggest that the callback is actually a pointer (which it is not, though it sort of works like one).